noir-projects/noir-contracts/contracts/schnorr_single_key_account_contract/src/util.nr

```diff
@@ -3,19 +3,19 @@ use dep::aztec::protocol_types::address::PublicKeysHash;
 use dep::std::{schnorr::verify_signature_slice};
 use crate::auth_oracle::AuthWitness;
 
-// TODO(#5830): the following is currently broken because we are no longer able to compute public keys hash
-// pub fn recover_address(message_hash: Field, witness: AuthWitness) -> AztecAddress {
-//     let message_bytes = message_hash.to_be_bytes(32);
-//     let verification = verify_signature_slice(
-//         witness.owner.x,
-//         witness.owner.y,
-//         witness.signature,
-//         message_bytes
-//     );
-//     assert(verification == true);
+pub fn recover_address(message_hash: Field, witness: AuthWitness) -> AztecAddress {
+    let message_bytes = message_hash.to_be_bytes(32);
+    // In a single key account contract we re-used ivpk_m as signing key
+    let verification = verify_signature_slice(
+        witness.ivpk_m.x,
+        witness.ivpk_m.y,
+        witness.signature,
+        message_bytes
+    );
+    assert(verification == true);
 
-//     AztecAddress::compute(
-//         PublicKeysHash::compute(witness.owner),
-//         witness.partial_address
-//     )
-// }
+    AztecAddress::compute(
+        PublicKeysHash::compute(witness.npk_m, witness.ivpk_m, witness.ovpk_m, witness.tpk_m),
```

Comment:
It seems a bit weird to me here if the viewing key is used as the signing key? But might just be the confusion over the "single_key" when the signing key is one of the other keys.

Reply:
Agree that it's weird. I created [an issue](https://github.com/AztecProtocol/aztec-packages/issues/5837) for this a while ago and it's linked [here](https://github.com/AztecProtocol/aztec-packages/blob/21c418a7eda84d5f0ebad049d419ed3709753cb8/yarn-project/circuits.js/src/keys/index.ts#L23).

It's like that currently because I wanted to minimize the number of changes when refactoring according to the new keys scheme and this was the most backwards compatible.